Firestore/core/src/firebase/firestore/api/write_batch.h

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_API_WRITE_BATCH_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_API_WRITE_BATCH_H_
+
+#if !defined(__OBJC__)
+#error "This header only supports Objective-C++"
+#endif  // !defined(__OBJC__)
+
+#import <Foundation/Foundation.h>
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "Firestore/core/src/firebase/firestore/core/user_data.h"
+#include "Firestore/core/src/firebase/firestore/util/status.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class FSTMutation;
+
+namespace firebase {
+namespace firestore {
+namespace api {
+
+class Firestore;
+class DocumentReference;
+
+class WriteBatch {
+ public:
+  WriteBatch() = delete;
```

Comment:
Very optional: defining a constructor prevents the default constructor from being generated, so strictly speaking, this is unnecessary. If you feel being more explicit adds readability here, feel free to keep as is.

Reply:
Ah! Right, thanks.